ci-operator/step-registry/rosa/cluster/wait-ready/cluster/rosa-cluster-wait-ready-cluster-commands.sh

```
@@ -174,8 +175,20 @@ while true; do
       break
     fi
   else
-      # Stall detection: fail early if state is stuck for too long
+      # Hive-stall detection: infra_id still null after entering 'installing' (Classic only)
       stall_elapsed=$(( current_time - dyn_start_time ))
+      if [[ "${CLUSTER_STATE}" == "installing" ]] && [[ "${HOSTED_CP}" != "true" ]] && (( stall_elapsed >= HIVE_START_TIMEOUT )); then
+        hive_infra_id=$(cat ${cluster_info_json} | jq -r '.infra_id')
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)